arrows/ocv/image_container.cxx

```diff
@@ -192,48 +197,63 @@ ::vital_to_ocv(const vital::image& img, image_container::ColorMode cm)
       out.addref();
     }
     // TODO use MatAllocator to share memory with image_memory
-    if(cm == RGB || out.channels() == 1 )
+    if( cm == RGB || cm == OTHER_COLOR_MODE || out.channels() < 3 || out.channels() > 4 )
     {
+      //Want output as something other than an BGR(A) image
       return out;
     }
     else
     {
-      cv::Mat bgr;
-      if ( out.channels() == 3 )
-      {
-        cv::cvtColor(out, bgr, CV_RGB2BGR);
-      }
-      else if ( out.channels() == 4 )
+      //Want output as a BGR(A) image, and it has the correct number of channels
+      switch(out.depth())
       {
-        cv::cvtColor(out, bgr, CV_RGBA2BGRA);
+        case CV_8U:
+        case CV_16U:
+        case CV_32F:
+        {
+          cv::Mat bgr;
+          cv::cvtColor(out, bgr, (out.channels() == 3)?CV_BGR2RGB:CV_BGRA2RGBA);
+          return bgr;
+        }
+        case CV_8S:
+        case CV_16S:
+        case CV_32S:
+        case CV_64F:
+          VITAL_THROW( image_type_mismatch_exception,"Only CV_8U, CV_16U, and CV_32F are supported for BGR and RGB conversion");
       }
-      return bgr;
     }
   }
 
   // allocated a new cv::Mat
   cv::Mat out(static_cast<int>(img.height()), static_cast<int>(img.width()),
               CV_MAKETYPE(cv_type, static_cast<int>(img.depth())));
   // wrap the new image as a VITAL image (always a shallow copy)
-  image new_img = ocv_to_vital(out);
+  image new_img = ocv_to_vital(out, RGB);
   new_img.copy_from(img);
 
-  if(cm == RGB || out.channels() == 1 )
+  if( cm == RGB || cm == OTHER_COLOR_MODE || out.channels() < 3 || out.channels() > 4 )
```

Comment:
It might be better to use `cm != BGR` instead of `cm == RGB || cm == OTHER_COLOR_MODE` both because it is more compact and also because the latter will break if more color modes are added in the future.